lib/from_openstudio/model.rb

```diff
@@ -178,5 +201,49 @@ def self.materials_from_model(openstudio_model)
       result
     end
 
+    # Create HB Construction from OpenStudio Materials
+    def self.constructions_from_model(openstudio_model)
+      result = []
+
+      # Create HB AirConstruction from OpenStudio Construction
+      openstudio_model.getConstructionAirBoundarys.each do |construction|
+        result << AirBoundaryConstructionAbridged.from_construction(construction)
+      end
+
+      # Create HB WindowConstruction from OpenStudio Construction
+      openstudio_model.getConstructions.each do |construction|
+        if construction.isFenestration
+          window_construction = false
+          construction.layers.each do |material|
```

Comment:
You don't need to do a whole loop here. Just grab the first/outermost material layer and check if it's Standard Glazing or Simple Glazing. It's illegal in E+ to have a gas gap on the outer layer of a window construction so you don't need to check for that material type if you are only checking the first layer.